tests/test_edge_cases.py

```diff
@@ -2446,3 +2446,60 @@ class C(BaseModel):
     assert exc_info.value.errors() == [
         {'input': C(), 'loc': ('B',), 'msg': 'Input should be a valid dictionary', 'type': 'dict_type'}
     ]
+
+
+@pytest.mark.skipif(sys.version_info < (3, 9), reason='cannot parametrize types before 3.9')
+@pytest.mark.parametrize(
+    ('sequence_type', 'input_data', 'expected_error_type', 'expected_error_msg'),
+    [
+        pytest.param(list[str], '1bc', 'list_type', 'Input should be a valid list', id='list[str]'),
+        pytest.param(
+            Sequence[str],
+            '1bc',
+            'sequence_str_str_type',
+            '`str` instances are not allowed as a Sequence value',
+            id='Sequence[str]',
+        ),
+        pytest.param(
+            Sequence[bytes],
+            b'1bc',
+            'sequence_str_str_type',
+            '`bytes` instances are not allowed as a Sequence value',
+            id='Sequence[bytes]',
+        ),
+    ],
+)
+def test_sequences_str(sequence_type, input_data, expected_error_type, expected_error_msg):
+    input_sequence = [input_data[:1], input_data[1:]]
+
+    str_sequence_analyzer: AnalyzedType[sequence_type] = AnalyzedType(sequence_type)
```

Comment:
you don't need both `AnalysedType` and a model, one is enough.

Reply:
Which one would be preferred?